PROTO_tests/tests/numeric_test.py

```diff
@@ -828,3 +832,169 @@ def test_putmask(self, prob_size):
             values = np.arange(10)
             with pytest.raises(TypeError):
                 ak.putmask(pda, pda > 3, values)
+
+    # In the tests below, the rationale for using size = math.sqrt(prob_size) is that
+    # the resulting matrices are on the order of size*size.
+
+    # tril works on ints, floats, or bool
+
+    @pytest.mark.parametrize("data_type", INT_FLOAT_BOOL)
+    @pytest.mark.parametrize("prob_size", pytest.prob_size)
+    def test_tril(self, data_type, prob_size):
+        if get_max_array_rank() < 2:
+            pytest.skip()
+
+        size = int(sqrt(prob_size))
+
+        # ints and bools are checked for equality; floats are checked for closeness
+
+        check = lambda a, b, t: (
+            np.allclose(a.tolist(), b.tolist()) if akdtype(t) == "float64" else (a == b).all()
+        )
+
+        # test on one square and two non-square matrices
+
+        for rows, cols in [(size, size), (size + 1, size - 1), (size - 1, size + 1)]:
+            pda = ak.randint(1, 10, (rows, cols))
+            nda = pda.to_ndarray()
+            sweep = range(-(rows - 2), cols)  # sweeps the diagonal from LL to UR
+            for diag in sweep:
+                npa = np.tril(nda, diag)
+                ppa = ak.tril(pda, diag).to_ndarray()
+                assert check(npa, ppa, data_type)
+
+    # triu works on ints, floats, or bool
+
+    @pytest.mark.parametrize("data_type", INT_FLOAT_BOOL)
+    @pytest.mark.parametrize("prob_size", pytest.prob_size)
+    def test_triu(self, data_type, prob_size):
+        if get_max_array_rank() < 2:
+            pytest.skip()
```

Comment:
The reason I changed it to this is that `get_max_array_rank()` cannot be called before the test because the arkouda server needs to connect first.  `get_server_max_array_dims()` depends on the config file which is not as accurate.

Reply:
Good suggestion, definitely.

Reply:
@ajpotts ah! well in that case, should we change other spots that use this `skipif` to do the same `get_max_array_rank` check? either way that would be outside the scope of this PR

Reply:
I made a separate ticket for this question: https://github.com/Bears-R-Us/arkouda/issues/3664